pkg/controllers/serviceaccount/serviceaaccount_controller.go

```diff
@@ -0,0 +1,283 @@
+func NewServiceAccountController(ctx context.Context, k8sClientset *kubernetes.Clientset, c client.Interface) controller.Controller {
+	serviceAccountConverter := converter.NewServiceAccountConverter()
+
+	// Function returns map of profile_name:object stored by policy controller
+	// in the Calico datastore. Identifies controller written objects by
+	// their naming convention.
+	listFunc := func() (map[string]interface{}, error) {
+		log.Debugf("Listing profiles from Calico datastore: to check for ServiceAccount")
+		filteredProfiles := make(map[string]interface{})
+
+		// Get all profile objects from Calico datastore.
+		profileList, err := c.Profiles().List(ctx, options.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		// Filter out only objects that are written by policy controller.
+		for _, profile := range profileList.Items {
+			if strings.HasPrefix(profile.Name, kdd.ServiceAccountProfileNamePrefix) {
+				// Update the profile's ObjectMeta so that it simply contains the name.
+				// There is other metadata that we might receive (like resource version) that we don't want to
+				// compare in the cache.
+				profile.ObjectMeta = metav1.ObjectMeta{Name: profile.Name}
+				key := serviceAccountConverter.GetKey(profile)
+				filteredProfiles[key] = serviceAccountProfile{profile}
+			}
+		}
+		log.Debugf("Found %d ServiceAccount profiles in Calico datastore", len(filteredProfiles))
+		return filteredProfiles, nil
+	}
+
+	// Create a Cache to store Profiles in.
+	cacheArgs := rcache.ResourceCacheArgs{
+		ListFunc:   listFunc,
+		ObjectType: reflect.TypeOf(serviceAccountProfile{}),
+	}
+	ccache := rcache.NewResourceCache(cacheArgs)
+
+	// Create a ServiceAccount watcher.
+	listWatcher := cache.NewListWatchFromClient(k8sClientset.Core().RESTClient(), "serviceaccounts", "", fields.Everything())
+
+	// Bind the calico cache to kubernetes cache with the help of an informer. This way we make sure that
+	// whenever the kubernetes cache is updated, changes get reflected in the Calico cache as well.
+	_, informer := cache.NewIndexerInformer(listWatcher, &v1.ServiceAccount{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			log.Debugf("Got ADD event for ServiceAccount: %#v", obj)
+			profile, err := serviceAccountConverter.Convert(obj)
+			if err != nil {
+				log.WithError(err).Errorf("Error while converting %#v to calico profile.", obj)
+				return
+			}
+
+			// Add to cache.
+			k := serviceAccountConverter.GetKey(profile)
+			ccache.Set(k, serviceAccountProfile{profile})
+		},
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			log.Debugf("Got UPDATE event for ServiceAccount")
+			log.Debugf("Old object: \n%#v\n", oldObj)
+			log.Debugf("New object: \n%#v\n", newObj)
+
+			// Convert the ServiceAccount into a Profile.
+			profile, err := serviceAccountConverter.Convert(newObj)
+			if err != nil {
+				log.WithError(err).Errorf("Error while converting %#v to calico profile.", newObj)
```

Comment:
nit: let's be consistent with capitalization of "Calico" in the logs.

Reply:
yes.